src/x86/uarch.c

```diff
@@ -167,6 +173,26 @@ enum cpuinfo_uarch cpuinfo_x86_decode_uarch(
 						case 0x7D: // Ice Lake-Y
 						case 0x7E: // Ice Lake-U
 							return cpuinfo_uarch_sunny_cove;
+						case 0xA7: // Rocket Lake
+							return cpuinfo_uarch_cypress_cove;
+						case 0x9A: // Alder Lake
```

Comment:
Alder Lake pcore can also have a model 0x97:
See Also this excerpt from https://stackoverflow.com/questions/69955410/how-to-detect-p-e-core-in-intel-alder-lake-cpu

The CPUID instruction gives information about the core, on which it is executed. It is different for P cores and E cores.

The CPUID on Alder Lake is family 6 model 0x9A for both cores when enabled. The CPUID is changed to family 6 model 0x97 when E cores are disabled and AVX512 is enabled.

CPUID leaf 7 EDX bit 15 indicates a hybrid design.

CPUID leaf 1A EAX bit 24-31 indicates the type of core, according to "Game Dev Guide for Alder Lake Performance Hybrid Architecture", https://www.intel.com/content/www/us/en/developer/articles/guide/alder-lake-developer-guide.html

See my discussion at https://www.agner.org/forum/viewtopic.php?f=1&t=79

Reply:
New model IDs have been added for Rocket, Alder, and Meteor Lake.